app/models/miq_server/worker_management/process.rb

```diff
@@ -1,13 +1,29 @@
 class MiqServer::WorkerManagement::Process < MiqServer::WorkerManagement
   def sync_from_system
     require "sys/proctable"
-    self.miq_processes = Sys::ProcTable.ps.select { |proc| proc.ppid == my_server.pid }
+    @miq_processes_by_pid = Sys::ProcTable.ps.select { |proc| proc.ppid == my_server.pid }.index_by(&:pid)
   end
 
   def sync_starting_workers
     MiqWorker.find_all_starting.to_a
   end
 
+  def cleanup_orphaned_worker_rows
+    orphaned_rows = miq_workers.where.not(:pid => miq_pids)
```

Comment:
We use system_uid column in kubernetes - is that column populated with pid in process mode?  If so, may be worth using the same column for consistency.

Reply:
It isn't set for process or systemd but we should do that for consistency.  That's going to be another PR but we can probably get that in first and make use of it here.